src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -84,20 +82,18 @@ protected override void OnFault(Exception e)
             base.OnFault(e);
         }
 
-        internal void CallFromNativeContract(Action onComplete, UInt160 hash, string method, params StackItem[] args)
+        internal void CallFromNativeContract(UInt160 callingScriptHash, UInt160 hash, string method, params StackItem[] args)
```

Comment:
I think we shouldn't modify the callingScriptHash. Because it may cause confusion.

Reply:
But we need for calling script hash to be correct for native invocations also. The problem now is that when native contracts call each other (natively), these calls are not reflected in any way on VM execution stack.